patches/net/minecraft/world/effect/PoisonMobEffect.java.patch

```diff
@@ -1,12 +1,16 @@
 --- a/net/minecraft/world/effect/PoisonMobEffect.java
 +++ b/net/minecraft/world/effect/PoisonMobEffect.java
-@@ -11,7 +_,8 @@
+@@ -11,7 +_,12 @@
      public void applyEffectTick(LivingEntity p_296276_, int p_296233_) {
          super.applyEffectTick(p_296276_, p_296233_);
          if (p_296276_.getHealth() > 1.0F) {
 -            p_296276_.hurt(p_296276_.damageSources().magic(), 1.0F);
 +            // Neo: Replace DamageSources#magic() with neoforge:poison to allow differentiating poison damage.
-+            p_296276_.hurt(p_296276_.damageSources().source(net.neoforged.neoforge.common.NeoForgeMod.POISON_DAMAGE), 1.0F);
++            // Fallback to minecraft:magic in client code when connecting to a vanilla server.
++            // LivingEntity#hurt(DamageSource) will no-op in client code immediately, but the holder is resolved before the no-op.
++            var dTypeReg = p_296276_.damageSources().damageTypes;
++            var dType = dTypeReg.getHolder(net.neoforged.neoforge.common.NeoForgeMod.POISON_DAMAGE).orElse(dTypeReg.getHolderOrThrow(net.minecraft.world.damagesource.DamageTypes.MAGIC));
++            p_296276_.hurt(new net.minecraft.world.damagesource.DamageSource(dType), 1.0F);
```

Comment:
Would it be better to provide `p_296276_.damageSources().magic()` for the fallback case instead of constructing a new `DamageSource`? Don't know if it matters.

Reply:
I don't *think* it matters. DS objects are fairly cheap, vanilla just caches them because they can.